sympy/solvers/ode.py

```diff
@@ -1424,6 +1424,8 @@ class in it.  Note that a hint may do this anyway if
         if r and not any(r[i].has(x) for i in r if i >= 0):
             # Inhomogeneous case: F(x) is not identically 0
             if r[-1]:
+                global eq_homogenous
```

Comment:
No global variables. I know that there are already some but they should be removed.

Reply:
then how should i access that eq_homogenous because when i tried by passing argument it conflicted with other methods too